app/src/main/java/com/example/habitsmasher/UserDatabaseHelper.java

```diff
@@ -97,14 +108,18 @@ private void setNumberOfFollowing(List<Object> following) {
      * @param followers the followers collection of the user
      */
     private void setNumberOfFollowers(List<Object> followers) {
-        if (followers != null && !followers.isEmpty()) {
-            if (followers.get(0) == "") {
-                _numberOfFollowers.setText("0");
+        try {
+            if (followers != null && !followers.isEmpty()) {
+                if (followers.get(0) == "") {
+                    _numberOfFollowers.setText("0");
+                } else {
+                    _numberOfFollowers.setText(String.valueOf(followers.size()));
+                }
             } else {
-                _numberOfFollowers.setText(String.valueOf(followers.size()));
+                _numberOfFollowers.setText("0");
             }
-        } else {
-            _numberOfFollowers.setText("0");
+        } catch (NullPointerException e) {
+            Log.d(TAG, "TextView for number followers not found");
```

Comment:
Nothing changed here except for the fact that this is now wrapped in a try/catch block. An NPE can occur if we are currently not on the profile page, so there's nothing to update. In this case, we don't need to do anything, just log a message for debugging.